specification/indices/put_index_template/IndicesPutIndexTemplateRequest.ts

```diff
@@ -181,4 +182,9 @@ export class IndexTemplateMapping {
    * @availability serverless stability=stable
    */
   lifecycle?: DataStreamLifecycle
+  /**
+   * @availability stack since=8.19.0 stability=stable
+   * @availability serverless stability=stable
+   */
+  data_stream_options?: DataStreamOptionsTemplate
```

Comment:
According to this failing test

```ts
// Test file: /test/free/data_stream/230_data_stream_options.yml
// Test name: Test resetting data stream options in template composition

import { expectAssignable } from 'tsd'
import * as T from '../types'

expectAssignable<T.IndicesPutIndexTemplateRequest>({
  "body": {
    "composed_of": [
      "my-component-template"
    ],
    "index_patterns": [
      "no-data-stream"
    ],
    "template": {
      "data_stream_options": null,
      "settings": {
        "number_of_replicas": 0
      }
    }
  },
  "name": "my-valid-template"
})
```

We need this change:

```suggestion
  data_stream_options?: DataStreamOptionsTemplate | null
```